### ISSUE TITLE
Poetry does not work on Python 3.5.2

### ISSUE THREAD

**Issue by cole** [python-poetry/poetry#1744]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: OS X, Python docker images
- **Poetry version**: 1.0.0
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://raw.githubusercontent.com/cole/aiosmtplib/master/pyproject.toml

## Issue
`poetry install` fails on version 3.5.2:

```
$ poetry install -v
Traceback (most recent call last):
  File "/root/.tox/py35-asyncio/bin/poetry", line 7, in <module>
    from poetry.console import main
  File "/root/.tox/py35-asyncio/lib/python3.5/site-packages/poetry/console/__init__.py", line 1, in <module>
    from .application import Application
  File "/root/.tox/py35-asyncio/lib/python3.5/site-packages/poetry/console/application.py", line 1, in <module>
    from cleo import Application as BaseApplication
  File "/root/.tox/py35-asyncio/lib/python3.5/site-packages/cleo/__init__.py", line 3, in <module>
    from .application import Application
  File "/root/.tox/py35-asyncio/lib/python3.5/site-packages/cleo/application.py", line 4, in <module>
    from clikit.console_application import ConsoleApplication
  File "/root/.tox/py35-asyncio/lib/python3.5/site-packages/clikit/__init__.py", line 1, in <module>
    from .api.config.application_config import ApplicationConfig
  File "/root/.tox/py35-asyncio/lib/python3.5/site-packages/clikit/api/config/__init__.py", line 1, in <module>
    from .application_config import ApplicationConfig
  File "/root/.tox/py35-asyncio/lib/python3.5/site-packages/clikit/api/config/application_config.py", line 6, in <module>
    from typing import ContextManager
ImportError: cannot import name 'ContextManager'
```

The cause of this issue seems to be in clikit. It's also tracked in that repo: https://github.com/sdispater/clikit/issues/7.

Likely an issue on 3.5.0-3.5.3, however 3.5.2 is the most commonly used of those versions.

**Comment by MadWombat** [comment]
Also, 3.5.2 is the default python on Ubuntu Xenial. Xenial is an LTS release with end of life in 2024, so python 3.5.2 will be around for a while.

**Comment by rvanlaar** [comment]
We're getting hit by this issue right now, what's that status of this issue? 

As in, how can I help to move this forward?

**Comment by rvanlaar** [comment]
Would a pull request help and be accepted, once CliKit has a new release?

**Comment by rvanlaar** [comment]
This issue in Clikit was merged today: https://github.com/sdispater/clikit/pull/8
This fix is included in the version 0.4.2 of clikit, that was just released.
`pip3 install poetry` works.

**Comment by cole** [comment]
Fixed now with the release of clikit 0.4.2.
